bchrpc/server.go

```diff
@@ -759,7 +798,8 @@ func (s *GrpcServer) GetAddressTransactions(ctx context.Context, req *pb.GetAddr
 	// Attempt to decode the supplied address.
 	addr, err := bchutil.DecodeAddress(req.Address, s.chainParams)
 	if err != nil {
-		return nil, status.Error(codes.InvalidArgument, "invalid address")
+		addr, err = goslp.DecodeAddress(req.Address, s.chainParams)
+		addr, err = goslp.ConvertSlpToCashAddress(addr, s.chainParams)
```

Comment:
So I see you falling back to trying to decode the slp address and convert it to a cash address, however I don't see you handling the error case where both fail. If your fallback fails we should be returning an invalid address like before.

Reply:
If we are doing this in any other spot we might want to write a custom decode function that encapsulates all this logic. It would make the error handling a bit prettier.

Reply:
done.